internal/routing/acquire.go

```diff
@@ -15,11 +16,27 @@ func acquire(w http.ResponseWriter, r *http.Request, ls *lockservice.SimpleLockS
 		return
 	}
 
-	desc := &lockservice.SimpleDescriptor{
-		FileID: string(body),
+	var req interface{}
+	err = json.Unmarshal(body, &req)
```

Comment:
Unmarshalling into interfaces when you know what to expect from the request is not a good idea. Have a definite type.

Reply:
https://stackoverflow.com/questions/35583735/unmarshaling-into-an-interface-and-then-performing-type-assertion

See the question to understand why I had to do this

Reply:
Yes thats what I meant when I commented above. Have a concrete type for communication always. It should not be changed. If in the future we change, we can change the Unmarshalling here.

Reply:
The logic has been changed. I am simply using the `type Request{}` struct now for Marshaling and Unmarshaling.